packages/sanity/src/core/form/inputs/CrossDatasetReferenceInput/useReferenceInfo.ts

```diff
@@ -42,27 +42,35 @@ export function useReferenceInfo(
     setRetryAttempt((current) => current + 1)
   }, [])
 
+  const docInfo = useMemo(() => ({_id: doc._id, _type: doc._type}), [doc._id, doc._type])
   const referenceInfo = useMemoObservable(
     () =>
-      doc._id
-        ? getReferenceInfo(doc).pipe(
+      docInfo._id
+        ? getReferenceInfo(docInfo).pipe(
             map(
               (result) =>
                 ({
                   isLoading: false,
                   result,
                   error: undefined,
                   retry,
+                  retryAttempt,
```

Comment:
Adding `retryAttempt` here makes it natural for the linter to treat it as part of the deps array.